File: Data Structures & Algorithms/islands-and-treasure/submission-0.cpp
class Solution {
public:
    int INF = 2147483647;
    void islandsAndTreasure(vector<vector<int>>& grid) {
        int m = grid.size();
        int n = grid[0].size();
        cout << m << "\n";
        cout << n << "\n";
        queue<tuple<int, int, int>> bfsqueue;
        for (int row = 0; row < m; row++) {
            for (int col = 0; col < n; col++) {
                if (grid[row][col] == 0) {
                    bfsqueue.push(make_tuple(row, col, 0));
                }
            }
        }
        while (not bfsqueue.empty()) {
            traverse(grid, m, n, bfsqueue);
        }
    }
    void traverse(vector<vector<int>>& grid, int m, int n, 
    queue<tuple<int, int, int>>& bfs) {
            int startrow = get<0>(bfs.front());
            int startcol = get<1>(bfs.front());
            int value = get<2>(bfs.front());
            cout << startrow << " " << startcol << " " << value << "\n";
            if (grid[startrow][startcol] == INF) {
                grid[startrow][startcol] = value;
            }
            if (startrow != 0 && grid[startrow-1][startcol] == INF) {
                bfs.push(make_tuple(startrow-1,startcol,value+1));
            }
            if (startrow != m-1 && grid[startrow+1][startcol] == INF) {
                bfs.push(make_tuple(startrow+1,startcol,value+1));
            }
            if (startcol != 0 && grid[startrow][startcol-1] == INF) {
                bfs.push(make_tuple(startrow,startcol-1,value+1));
            }
            if (startcol != n-1 && grid[startrow][startcol+1] == INF) {
                bfs.push(make_tuple(startrow,startcol+1,value+1));
            }
            bfs.pop();
    };
};

File: Data Structures & Algorithms/rotting-fruit/submission-0.py
class Solution:
    def orangesRotting(self, grid: List[List[int]]) -> int:
        m = len(grid)
        n = len(grid[0])
        fresh = 0
        queue = []
        days = 0
        #count no. of fresh fruits
        for row in range(m):
            for col in range(n):
                if (grid[row][col] == 1):
                    fresh += 1
                    print(fresh)
                if (grid[row][col] == 2):
                    queue.append((row, col, 0))
        #simulated rotting
        while (fresh and queue):
            (fresh, days) = self.traverse(grid, queue, fresh, days, m, n)
        if fresh:
            return -1
        else:
            return days
    def traverse(self, grid: List[List[int]], queue: List[Tuple(int, int, int)], fresh: int, days: int, m: int, n: int):
        (row, col, value) = queue[0]
        print(str(row) + " " + str(col) + " " + str(value))
        if (grid[row][col] == 1):
            grid[row][col] = 2
            fresh = fresh - 1
            #print(str(row) + " " + str(col) + " " + str(value))
            print(fresh)
            if (value > days):
                days = value
                print(str(days) + "days")
        if (row != 0 and grid[row-1][col] == 1):
            queue.append((row-1, col, value+1))
        if (row != m-1 and grid[row+1][col] == 1):
            queue.append((row+1, col, value+1))
        if (col != 0 and grid[row][col-1] == 1):
            queue.append((row, col-1, value+1))
        if (col != n-1 and grid[row][col+1] == 1):
            queue.append((row, col+1, value+1))
        queue.pop(0)
        return (fresh, days)
        

                
        